src/main/io/osd.c

```diff
@@ -255,6 +268,89 @@ static void osdFormatCoordinate(char *buff, char sym, int32_t val)
     tfp_sprintf(buff + 1, "%s.%s", wholeDegreeString, wholeUnshifted + strlen(wholeDegreeString));
 }
 
+static const char * osdArmingDisabledReasonMessage(void)
+{
+    switch (isArmingDisabledReason()) {
+        case ARMING_DISABLED_FAILSAFE_SYSTEM:
+            return OSD_MESSAGE_STR("FAILSAFE SYSTEM IS ACTIVE");
+        case ARMING_DISABLED_NOT_LEVEL:
+            return OSD_MESSAGE_STR("AIRCRAFT IS NOT LEVEL");
+        case ARMING_DISABLED_SENSORS_CALIBRATING:
+            return OSD_MESSAGE_STR("SENSORS CALIBRATING");
+        case ARMING_DISABLED_SYSTEM_OVERLOADED:
+            return OSD_MESSAGE_STR("SYSTEM OVERLOADED");
+        case ARMING_DISABLED_NAVIGATION_UNSAFE:
+            return OSD_MESSAGE_STR("NAVIGATION IS UNSAFE");
+        case ARMING_DISABLED_COMPASS_NOT_CALIBRATED:
+            return OSD_MESSAGE_STR("COMPASS NOT CALIBRATED");
+        case ARMING_DISABLED_ACCELEROMETER_NOT_CALIBRATED:
+            return OSD_MESSAGE_STR("ACCELEROMETER NOT CALIBRATED");
+        case ARMING_DISABLED_ARM_SWITCH:
+            return OSD_MESSAGE_STR("DISABLE ARM SWITCH FIRST");
+        case ARMING_DISABLED_HARDWARE_FAILURE:
+            {
+                if (!HW_SENSOR_IS_HEALTHY(getHwGyroStatus())) {
+                    return OSD_MESSAGE_STR("GYRO FAILURE");
+                }
+                if (!HW_SENSOR_IS_HEALTHY(getHwAccelerometerStatus())) {
+                    return OSD_MESSAGE_STR("ACCELEROMETER FAILURE");
+                }
+                if (!HW_SENSOR_IS_HEALTHY(getHwCompassStatus())) {
+                    return OSD_MESSAGE_STR("COMPASS FAILURE");
+                }
+                if (!HW_SENSOR_IS_HEALTHY(getHwBarometerStatus())) {
+                    return OSD_MESSAGE_STR("BAROMETER FAILURE");
+                }
+                if (!HW_SENSOR_IS_HEALTHY(getHwGPSStatus())) {
+                    return OSD_MESSAGE_STR("GPS FAILURE");
+                }
+                if (!HW_SENSOR_IS_HEALTHY(getHwRangefinderStatus())) {
+                    return OSD_MESSAGE_STR("RANGE FINDER FAILURE");
+                }
+                if (!HW_SENSOR_IS_HEALTHY(getHwPitotmeterStatus())) {
+                    return OSD_MESSAGE_STR("PITOT METER FAILURE");
+                }
+            }
+            return OSD_MESSAGE_STR("HARDWARE FAILURE");
+        case ARMING_DISABLED_BOXFAILSAFE:
+            return OSD_MESSAGE_STR("FAILSAFE MODE ENABLED");
+        case ARMING_DISABLED_BOXKILLSWITCH:
+            return OSD_MESSAGE_STR("KILLSWITCH MODE ENABLED");
+        case ARMING_DISABLED_RC_LINK:
+            return OSD_MESSAGE_STR("NO RC LINK");
+        case ARMING_DISABLED_THROTTLE:
+            return OSD_MESSAGE_STR("THROTTLE IS NOT LOW");
+        case ARMING_DISABLED_CLI:
+            return OSD_MESSAGE_STR("CLI IS ACTIVE");
+        case ARMING_DISABLED_CMS_MENU:
```

Comment:
This would probably be never shown.

Reply:
Roger that! I wasn't quite sure of all the code paths that might set those bits. I'll remove the messages for `ARMING_DISABLED_CMS_MENU ` and `ARMING_DISABLED_OSD_MENU`